frontend/public/components/dashboard/inventory-card/inventory-item.tsx

```diff
@@ -45,102 +54,161 @@ const getStatusGroupIcons = (flags: FlagsObject) => {
-          <span className="co-dashboard-icon">{groupIcon}</span>
-          <span className="co-inventory-card__status-text">{count}</span>
-        </Link>
-      </div>
-    );
+const StatusLink = connectToFlags<StatusLinkProps>(
+  ...getFlagsForExtensions(plugins.registry.getDashboardsInventoryItemGroups()),
+)(React.memo(({ groupID, count, statusIDs, kind, namespace, filterType, flags }) => {
+  if (groupID === InventoryStatusGroup.NOT_MAPPED || !count) {
+    return null;
   }
+  const statusItems = encodeURIComponent(statusIDs.join(','));
+  const namespacePath = namespace ? `ns/${namespace}` : 'all-namespaces';
+  const to = filterType && statusItems.length > 0 ? `/k8s/${namespacePath}/${kind.plural}?rowFilter-${filterType}=${statusItems}` : `/k8s/${namespacePath}/${kind.plural}`;
+  const statusGroupIcons = getStatusGroupIcons(flags);
+  const groupIcon = statusGroupIcons[groupID] || statusGroupIcons[InventoryStatusGroup.NOT_MAPPED];
+  return (
+    <div className="co-inventory-card__status">
+      <Link to={to} style={{textDecoration: 'none'}}>
+        <span className="co-dashboard-icon">{groupIcon}</span>
+        <span className="co-inventory-card__status-text">{count}</span>
+      </Link>
+    </div>
+  );
+}));
+
+const ResourceTitleComponent = ({ kind, namespace, children }) => (
+  <Link to={resourcePathFromModel(kind, null, namespace)}>
+    {children}
+  </Link>
 );
 
 export const ResourceInventoryItem = connectToFlags<ResourceInventoryItemProps>(
   ...getFlagsForExtensions(plugins.registry.getDashboardsInventoryItemGroups()),
-)(React.memo(
-  ({ kind, useAbbr, resources, additionalResources, isLoading, mapper, namespace, error, showLink = true, flags = {}}) => {
-    const groups = mapper(resources, additionalResources);
-    const [singularTitle, pluralTitle] = useAbbr ? [kind.abbr, `${kind.abbr}s`] : [kind.label, kind.labelPlural];
-    return (
-      <InventoryItem
-        isLoading={isLoading}
-        singularTitle={singularTitle}
-        pluralTitle={pluralTitle}
-        count={resources.length}
-        error={error}
-      >
-        {Object.keys(groups).filter(key => groups[key].count > 0).map(key => showLink ?
-          (
-            <StatusLink
-              key={key}
-              kind={kind}
-              namespace={namespace}
-              groupID={key}
-              count={groups[key].count}
-              statusIDs={groups[key].statusIDs}
-              filterType={groups[key].filterType}
-              flags={flags}
-            />
-          ) : (
-            <Status
-              groupID={key}
-              count={groups[key].count}
-              flags={flags}
-            />
-          )
-        )}
-      </InventoryItem>
-    );
-  }
-));
+)(({
+  kind,
+  useAbbr,
+  resources = [],
+  additionalResources,
+  isLoading,
+  mapper,
+  namespace,
+  error,
+  showLink = true,
+  flags = {},
+  ExpandedComponent,
+}) => {
+  const TitleComponent = React.useCallback((props) => (
```

Comment:
This is interesting, why `React.useCallback` and not `React.memo` here?

Is it because of explicit dependency control declared through array `[kind, namespace]` ?

Reply:
I can define dependencies for `React.memo` too. But `React.memo` returns value of memoized function, `React.useCallback` returns memoized function (not its value)

Reply:
Great, thanks for explaining this.